react/SearchContext.js

```diff
@@ -1,169 +1,148 @@
-    params: PropTypes.shape({
-      category: PropTypes.string,
-      department: PropTypes.string,
-      term: PropTypes.string,
-    }),
-    /** Render runtime context */
-    runtime: PropTypes.shape({
-      page: PropTypes.string.isRequired,
-    }),
-    /** Query params */
-    query: PropTypes.shape({
-      map: PropTypes.string,
-      rest: PropTypes.string,
-      order: PropTypes.oneOf(SORT_OPTIONS.map(o => o.value)),
-      priceRange: PropTypes.string,
-    }),
-    /** Custom query `query` param */
-    queryField: PropTypes.string,
-    /** Custom query `map` param */
-    mapField: PropTypes.string,
-    /** Custom query `rest` param */
-    restField: PropTypes.string,
-    /** Custom query `orderBy` param */
-    orderByField: PropTypes.string,
-    /** Current extension point name */
-    nextTreePath: PropTypes.string,
-    /** Component to be rendered */
-    children: PropTypes.node.isRequired,
-    /** Max items to show per result page */
-    maxItemsPerPage: PropTypes.number.isRequired,
-  }
-
-  static defaultProps = {
-    maxItemsPerPage: DEFAULT_MAX_ITEMS_PER_PAGE,
-  }
+const SearchContext = ({
+  nextTreePath,
+  params,
+  maxItemsPerPage = DEFAULT_MAX_ITEMS_PER_PAGE,
+  queryField,
+  mapField,
+  orderByField,
+  query: {
+    order: orderBy = orderByField || SORT_OPTIONS[0].value,
+    page: pageQuery,
+    map: mapQuery,
+    priceRange,
+  },
+  children,
+  ...props
+}) => {
+  const { page: runtimePage } = useRuntime()
+  const map = mapQuery || createInitialMap(params)
+  const page = pageQuery ? parseInt(pageQuery) : DEFAULT_PAGE
+  const from = (page - 1) * maxItemsPerPage
+  const to = from + maxItemsPerPage - 1
 
-  static schema = {
-    title: 'editor.product-search.title',
-    type: 'object',
-    properties: {
-      maxItemsPerPage: {
-        title: 'editor.product-search.maxItemsPerPage',
-        type: 'number',
-      },
-      queryField: {
-        title: 'Query',
-        type: 'string',
-      },
-      mapField: {
-        title: 'Map',
-        type: 'string',
-      },
-      restField: {
-        title: 'Other Query Strings',
-        type: 'string',
-      },
-      orderByField: {
-        title: 'Order by field',
-        type: 'string',
-        default: SORT_OPTIONS[0].value,
-        enum: SORT_OPTIONS.map(opt => opt.value),
-        enumNames: SORT_OPTIONS.map(opt => opt.label),
-      },
-    },
-  }
+  const includeFacets = (map, query) =>
+    !!(map && map.length > 0 && query && query.length > 0)
 
-  render() {
-    const {
-      nextTreePath,
-      params,
-      maxItemsPerPage,
-      queryField,
-      mapField,
-      restField,
-      orderByField,
-      query: {
-        order: orderBy = orderByField || SORT_OPTIONS[0].value,
-        page: pageQuery,
-        map: mapQuery,
-        rest = '',
-        priceRange,
-      },
-      runtime: { page: runtimePage },
-      ...props
-    } = this.props
+  const query = Object.values(params)
```

Comment:
I have been told that this Object.values is not available in all browsers, may be better to use ramda's `values`

Reply:
render includes a [polyfill](https://github.com/vtex/render-server/blob/327a99cd663b515df9a8bd1cae820e1046dda042/node/utils/assetLoader.ts#L42) for it